windows/src/engine/keyman32/kmprocess.cpp

```diff
@@ -153,14 +153,14 @@ BOOL ProcessHook()
         return FALSE;
       }
       km_kbp_context_items_dispose(citems);
-      SendDebugMessageFormat(0, sdmGlobal, 0, "ProcessEvent: vkey[%d] ShiftState[%d] isDown[%d]", _td->state.vkey,
-                                                    static_cast<uint16_t>(Globals::get_ShiftState() & K_MODIFIERFLAG),
-                                                    (uint8_t)_td->state.isDown);
-      if (KM_KBP_STATUS_OK !=
-        (km_kbp_status_codes)km_kbp_process_event(_td->lpActiveKeyboard->lpCoreKeyboardState, _td->state.vkey,
-                                  static_cast<uint16_t>(Globals::get_ShiftState() & K_MODIFIERFLAG),
-                                                    (uint8_t)_td->state.isDown)) {
-        SendDebugMessageFormat(0, sdmGlobal, 0, "ProcessEvent CoreProcessEvent Result:False %d ",FALSE);
+      SendDebugMessageFormat(
+          0, sdmGlobal, 0, "ProcessEvent: vkey[%d] ShiftState[%d] isDown[%d]", _td->state.vkey,
+          static_cast<uint16_t>(Globals::get_ShiftState()), (uint8_t)_td->state.isDown);
+      if (KM_KBP_STATUS_OK != (km_kbp_status_codes)km_kbp_process_event(
+                                  _td->lpActiveKeyboard->lpCoreKeyboardState, _td->state.vkey,
+                                  static_cast<uint16_t>(Globals::get_ShiftState()),
```

Comment:
We need to be sure that passing in the unadulaterated shift state mask doesn't have side-effects elsewhere in `km_kbp_process_event`! Have you verified this?

Reply:
Only in the fact that it was like that up until around a week ago when I added in the `K_MODIFIERFLAG`. 

How about this working of `keyboardprocessor_vkeys.h`  and the   `km_kbp_modifier_state` enum (which has dead code of the scroll lock etc by the way). 

``` 
//  Mask out the bits not supported according to `km_kbp_modifier_state` enum, Update the mask if this enum is expanded.
 if (KM_KBP_STATUS_OK != (km_kbp_status_codes)km_kbp_process_event(
                                  _td->lpActiveKeyboard->lpCoreKeyboardState, _td->state.vkey,
                                  static_cast<uint16_t>(Globals::get_ShiftState() & (KM_KBP_MODIFIER_MASK_ALL | KM_KBP_MODIFIER_MASK_CAPS) ),
```

Reply:
Yeah, I like this -- it's a more targeted fix and means `km_kbp_process_event` never receives the num lock and scroll lock modifiers which it doesn't actually support -- and so then we don't need to change `VkToChar` :grin:

Reply:
Oh ok I was going to make VkeyToChar handle a broken pre-condition. However, if the core api is honoured it shouldn't be a problem.

Reply:
Yep. We could add a debug-only assertion as an input condition to `km_kbp_process_event` if we want to be pure.